Unity/Assets/Scripts/SCHIZO/Utilities/T2DArray.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using TriInspector;
+using UnityEngine;
+
+[CreateAssetMenu(menuName ="SCHIZO/Utilities/Texture2DArray")]
+public class T2DArray : ScriptableObject
+{
+    [InfoBox("All Textures must have same dimensions!", TriMessageType.Error, nameof(texturesDimensionsMismatch))]
+    [InfoBox("Please assign at least 2 textures, or use Texture2D if single texture is intended", TriMessageType.Warning, nameof(tooFewTextures))]
+    [Required]
+    public List<Texture> textures;
+
+    private bool texturesDimensionsMismatch = true;
+    private bool tooFewTextures = true;
+
+    private void OnValidate()
+    {
+        if (textures.Count > 1 ) ClearEmpty(textures);
+        tooFewTextures = textures.Count < 2;
+
+        if (textures.Count > 1 && textures[0] != null)
+        {
+            texturesDimensionsMismatch = false;
+            int w = textures[0].width;
+            int h = textures[0].height;
+
+            foreach (Texture texture in textures)
+            {
+                if (texture.width != w || texture.height != h)
+                {
+                    texturesDimensionsMismatch = true;
+                    return;
+                }
+            }
+        }
+    }
+
+    private List<T> ClearEmpty <T>(List<T> list)
+    {
+        list.RemoveAll(o => o.GetType() == typeof(T));

```

Comment:
uuh im not sure what this is supposed to be doing, but it's definitely not removing the empty values (it's removing all Textures that are not Texture2D or Texture2DArray, and will in fact throw an exception if there is an empty value)

Reply:
fixed by doing it in-place (i spent some time figuring out how to remove nulls from list and came up with that. the next day when i woke up and looked at it again i got confused as well, but for some reason it worked as intended but not the other way )